### PR TITLE
72 osgi cdi integration

### PR DESCRIPTION
**Pull request by jhinkey**
Hi Ray,

Would you review these three articles on OSGi CDI integration? They're in this last commit:

https://github.com/rotty3000/liferay-docs/commit/c2deb64fa2e25aa682a42f86af7b50da9f51fd9b

Thanks!

### CODE REVIEW

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/00-osgi-cdi-integration-intro.markdown [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+to customize @product@ components. 
+
+- Use OSGi services in beans: Leverage any OSGi service published on @product@ 
+in any bean. 
+
+The following use cases provide more detail. 
+
+## Use Case: Registering a CDI bean as an OSGi service 
+
+As a CDI bean developer, you want to use a CDI bean to customize or override an
```
Comment (rotty3000) [inline developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/00-osgi-cdi-integration-intro.markdown]: can we change `customize` with `provide`?

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/01-publishing-cdi-beans-as-osgi-services.markdown [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+OSGi CDI Integration lets you publish CDI beans as OSGi services, making them
+accessible via the Liferay's OSGi service registry. Here's how to publish your
+bean as a service:
+
+1.  Add a project dependency on the OSGi CDI Integration artifact. For example, 
+here's the dependency to use in a Maven `pom.xml` file:
+
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.service.cdi</artifactId>
```
Comment (rotty3000) [inline developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/01-publishing-cdi-beans-as-osgi-services.markdown]: should you add `<version>1.0.0</version>` to make this snippet copy & paste ready?

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/02-using-osgi-services-in-a-bean.markdown [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Using OSGi Services in a Bean 
+
+Any bean can use the `@org.osgi.service.cdi.annotations.Reference` annotation to
```
Comment (rotty3000) [inline developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/02-using-osgi-services-in-a-bean.markdown]: in the provide a service use case you don't use the FQCN. I would try to be consistent. Either use FQCN or not :)

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/02-using-osgi-services-in-a-bean.markdown [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+inject OSGi services. It's the easiest way for a bean to access an OSGi service. Here's how to do it:
+
+1.  Add a project dependency on the OSGi CDI Integration artifact. For example, 
+here's the dependency to use in a Maven `pom.xml` file:
+
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.service.cdi</artifactId>
```
Comment (rotty3000) [inline developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/02-using-osgi-services-in-a-bean.markdown]: version?

**Review by rotty3000 — CHANGES_REQUESTED** [review]
I don't remember seeing link to the OSGi CDI Integration Specification or CDI 2.0 specs anywhere.

https://osgi.org/specification/osgi.enterprise/7.0.0/service.cdi.html
http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/02-using-osgi-services-in-a-bean.markdown (hunk 3 above)
Comment (jhinkey) [reply developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/02-using-osgi-services-in-a-bean.markdown]: Used FQCN consistently now. Thanks.

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/01-publishing-cdi-beans-as-osgi-services.markdown (hunk 2 above)
Comment (jhinkey) [reply developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/01-publishing-cdi-beans-as-osgi-services.markdown]: Added version. Thanks

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/00-osgi-cdi-integration-intro.markdown (hunk 1 above)
Comment (jhinkey) [reply developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/00-osgi-cdi-integration-intro.markdown]: Changed `customize` to `provide`.

File: developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/00-osgi-cdi-integration-intro.markdown [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# OSGi CDI Integration 
+
+OSGi CDI Integration brings features and capabilities of Contexts and Dependency
+Injection (CDI) to OSGi and makes OSGi services available to beans.  Here's what
+you'll learn how to do:
+
+- Publish CDI beans as OSG services: Register CDI beans as services you can use
```
[the remark this answers is not in the document]
Comment (jhinkey) [reply developer/customization/articles/02-architecture/03-understanding-services/02-osgi-cdi-integration/00-osgi-cdi-integration-intro.markdown]: OSGi

**Comment by jhinkey** [comment]
Thanks Ray. I took the suggestions and corrections from your review.

**Pull request merged**